lib/jekyll/commands/serve.rb

```diff
@@ -99,9 +103,9 @@ def webrick_opts(opts)
         # Recreate NondisclosureName under utf-8 circumstance
 
         private
-        def file_handler_opts
+        def file_handler_opts(opts)
           WEBrick::Config::FileHandler.merge({
-            :FancyIndexing     => true,
+            :FancyIndexing     => opts['show_dir_listing'] ? true : false,
```

Comment:
Right on, @envygeeks, I don't like it in production either and as I said, I'm planning on reverting it later.